resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/submit/steps/DriverInitContainerBootstrapStepSuite.scala

```diff
@@ -0,0 +1,159 @@
+      initContainerSteps,
+      CONFIG_MAP_NAME,
+      CONFIG_MAP_KEY)
+
+    val preparedDriverSpec = bootstrapStep.configureDriver(baseDriverSpec)
+
+    assert(preparedDriverSpec.driverPod.getMetadata.getLabels.asScala ===
+      FirstTestInitContainerConfigurationStep$.additionalLabels)
+    val additionalDriverEnv = preparedDriverSpec.driverContainer.getEnv.asScala
+    assert(additionalDriverEnv.size === 1)
+    assert(additionalDriverEnv.head.getName ===
+      FirstTestInitContainerConfigurationStep$.additionalMainContainerEnvKey)
+    assert(additionalDriverEnv.head.getValue ===
+      FirstTestInitContainerConfigurationStep$.additionalMainContainerEnvValue)
+
+    assert(preparedDriverSpec.otherKubernetesResources.size === 2)
+    assert(preparedDriverSpec.otherKubernetesResources.contains(
+      FirstTestInitContainerConfigurationStep$.additionalKubernetesResource))
+    assert(preparedDriverSpec.otherKubernetesResources.exists {
+      case configMap: ConfigMap =>
+        val hasMatchingName = configMap.getMetadata.getName == CONFIG_MAP_NAME
+        val configMapData = configMap.getData.asScala
+        val hasCorrectNumberOfEntries = configMapData.size == 1
+        val initContainerPropertiesRaw = configMapData(CONFIG_MAP_KEY)
+        val initContainerProperties = new Properties()
+        Utils.tryWithResource(new StringReader(initContainerPropertiesRaw)) {
+          initContainerProperties.load(_)
+        }
+        val initContainerPropertiesMap = Maps.fromProperties(initContainerProperties).asScala
+        val expectedInitContainerProperties = Map(
+          SecondTestInitContainerConfigurationStep$.additionalInitContainerPropertyKey ->
+            SecondTestInitContainerConfigurationStep$.additionalInitContainerPropertyValue)
+        val hasMatchingProperties = initContainerPropertiesMap == expectedInitContainerProperties
+        hasMatchingName && hasCorrectNumberOfEntries && hasMatchingProperties
+      case _ => false
+    })
+
+    val initContainers = preparedDriverSpec.driverPod.getSpec.getInitContainers
+    assert(initContainers.size() === 1)
+    val initContainerEnv = initContainers.get(0).getEnv.asScala
+    assert(initContainerEnv.size === 1)
+    assert(initContainerEnv.head.getName ===
+      SecondTestInitContainerConfigurationStep$.additionalInitContainerEnvKey)
+    assert(initContainerEnv.head.getValue ===
+      SecondTestInitContainerConfigurationStep$.additionalInitContainerEnvValue)
+
+    val expectedSparkConf = Map(
+      INIT_CONTAINER_CONFIG_MAP_NAME.key -> CONFIG_MAP_NAME,
+      INIT_CONTAINER_CONFIG_MAP_KEY_CONF.key -> CONFIG_MAP_KEY,
+      SecondTestInitContainerConfigurationStep$.additionalDriverSparkConfKey ->
+        SecondTestInitContainerConfigurationStep$.additionalDriverSparkConfValue)
+    assert(preparedDriverSpec.driverSparkConf.getAll.toMap === expectedSparkConf)
+  }
+}
+
+private object FirstTestInitContainerConfigurationStep$ extends InitContainerConfigurationStep {
```

Comment:
Do we need `$` at the end of the object name?

Reply:
No, it shouldn't be there. Removed.